pkg/utils/filewatcher.go

```diff
@@ -0,0 +1,84 @@
+package utils
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// FileWatcher is an utility that
+type FileWatcher struct {
+	w    *fsnotify.Watcher
+	path string
+
+	writeGracePeriod time.Duration
+	timer            *time.Timer
+}
+
+func NewFileWatcher(path string) (*FileWatcher, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	return &FileWatcher{w: watcher, path: path, writeGracePeriod: 200 * time.Millisecond}, nil
+}
+
+func (fw *FileWatcher) Start(changeHandler func()) error {
+	// Ensure that the target that we're watching is not a symlink as we won't get any events when we're watching
+	// a symlink.
+	fileRealPath, err := filepath.EvalSymlinks(fw.path)
+	if err != nil {
+		return fmt.Errorf("adding watcher failed: %s", err)
+	}
```

Comment:
Upon testing i found out that the current hot-reload functionality of `/etc/hosts` are not working in mac because `/etc` is a symlink to `/private/etc`. 

When we're watching `/etc` we won't get any events whenever `/private/etc` changed.